In [1]:
import cPickle
def load_cpkl(path_to_cpkl_file):
    return cPickle.load(open(path_to_cpkl_file, "r"))
# end

# Exploration of Test file

In [13]:
from collections import Counter
test_path_to_cpkl_file = "/home/mg/workspace_mg/Interaction_Predictions/fout_protein_interface_prediction_using_graph_convolutional_networks/pl_data/test.cpkl"
test = load_cpkl(path_to_cpkl_file=test_path_to_cpkl_file)

# Test example 1 on Complex: `4GXU`

In [15]:
complex_code = test[1][10]["complex_code"]
print("r_vertex shape: {} for complex code: {}".format(test[1][10]["r_vertex"].shape, complex_code))
print("l_vertex shape: {} for complex code: {}".format(test[1][10]["l_vertex"].shape, complex_code))
print("r_edge shape: {} for complex code: {}".format(test[1][10]["r_edge"].shape, complex_code))
print("l_edge shape: {} for complex code: {}".format(test[1][10]["l_edge"].shape, complex_code))
print("r_hood_indices shape: {} for complex code: {}".format(test[1][10]["r_hood_indices"].shape, complex_code))
print("l_hood_indices shape: {} for complex code: {}".format(test[1][10]["l_hood_indices"].shape, complex_code))
print("label shape: {} for complex code: {}".format(test[1][10]["label"].shape, complex_code))
print("expected dimensions for axis 1 for 'label' numpy array: {}".format(test[1][10]["r_vertex"].shape[0]*test[1][10]["l_vertex"].shape[0]))
print("actual dimensions for axis 1 for 'label' numpy array: {}".format(test[1][10]["label"].shape[0]))
print("label distribution: {}".format(Counter(test[1][10]["label"][:,2])))

r_vertex shape: (440, 70) for complex code: 4GXU
l_vertex shape: (1452, 70) for complex code: 4GXU
r_edge shape: (440, 20, 2) for complex code: 4GXU
l_edge shape: (1452, 20, 2) for complex code: 4GXU
r_hood_indices shape: (440, 20, 1) for complex code: 4GXU
l_hood_indices shape: (1452, 20, 1) for complex code: 4GXU
label shape: (636126, 3) for complex code: 4GXU
expected dimensions for axis 1 for 'label' numpy array: 638880
actual dimensions for axis 1 for 'label' numpy array: 636126
label distribution: Counter({-1: 636012, 1: 114})


# complex_code `4GXU`
Searched for more information on pdb website https://www.rcsb.org/structure/4GXU to confirm length for receptor and ligand structures.
## Entity ID: 1
* Molecule: Hemagglutinin HA1 chain
* Chains: A, C, E, G, I, K
* Sequence Length: 331
* Organism: Influenza A virus (strain A/Brevig Mission/1/1918 H1N1)
* Details: Mutation(s): 0; Gene Names: HA

## Entity ID: 2
* Molecule: Hemagglutinin HA2 chain
* Chains: B, D, F, H, J, L
* Sequence Length: 176
* Organism: Influenza A virus (strain A/Brevig Mission/1/1918 H1N1)
* Details: Mutation(s): 0; Gene Names: HA

## Entity ID: 3
* Molecule: Antibody 1F1, heavy chain
* Chains: M, O, Q, S, U, W
* Sequence Length: 231
* Organism: N/A
* Details: Mutation(s): 0

## Entity ID: 4
* Molecule: Antibody 1F1, light chain
* Chains: N, P, R, T, V, X
* Sequence Length: 217
* Organism: N/A
* Details: Mutation(s): 0

None of the lengths found on pdb website for complex `4GXU` seem to have a ligand or receptor length that matches what is found in the numpy array provided.

# Test example 2 on Complex: `4GAM`

In [19]:
complex_code = test[1][1]["complex_code"]
print("r_vertex shape: {} for complex code: {}".format(test[1][1]["r_vertex"].shape, complex_code))
print("l_vertex shape: {} for complex code: {}".format(test[1][1]["l_vertex"].shape, complex_code))
print("r_edge shape: {} for complex code: {}".format(test[1][1]["r_edge"].shape, complex_code))
print("l_edge shape: {} for complex code: {}".format(test[1][1]["l_edge"].shape, complex_code))
print("r_hood_indices shape: {} for complex code: {}".format(test[1][1]["r_hood_indices"].shape, complex_code))
print("l_hood_indices shape: {} for complex code: {}".format(test[1][1]["l_hood_indices"].shape, complex_code))
print("label shape: {} for complex code: {}".format(test[1][1]["label"].shape, complex_code))
print("expected dimensions for axis 1 for 'label' numpy array: {}".format(test[1][1]["r_vertex"].shape[0]*test[1][1]["l_vertex"].shape[0]))
print("actual dimensions for axis 1 for 'label' numpy array: {}".format(test[1][1]["label"].shape[0]))
print("label distribution: {}".format(Counter(test[1][1]["label"][:,2])))

r_vertex shape: (2128, 70) for complex code: 4GAM
l_vertex shape: (141, 70) for complex code: 4GAM
r_edge shape: (2128, 20, 2) for complex code: 4GAM
l_edge shape: (141, 20, 2) for complex code: 4GAM
r_hood_indices shape: (2128, 20, 1) for complex code: 4GAM
l_hood_indices shape: (141, 20, 1) for complex code: 4GAM
label shape: (279472, 3) for complex code: 4GAM
expected dimensions for axis 1 for 'label' numpy array: 300048
actual dimensions for axis 1 for 'label' numpy array: 279472
label distribution: Counter({-1: 279104, 1: 368})


# complex_code `4GAM`
Searched for more information on pdb website https://www.rcsb.org/structure/4GAM to confirm length for receptor and ligand structures.

## Entity ID: 1
* Molecule: Methane monooxygenase component A beta chain
* Chains: B, G, L, Q
* Sequence Length: 389
* Organism: Methylococcus capsulatus (strain ATCC 33009 / NCIMB 11132 / Bath)
* Details: Mutation(s): 0; Gene Names: mmoY; EC: 1.14.13.25

## Entity ID: 2
* Molecule: Methane monooxygenase component A alpha chain	
* Chains: A, F, K, P
* Sequence Length: 527
* Organism: Methylococcus capsulatus (strain ATCC 33009 / NCIMB 11132 / Bath)
* Details: Mutation(s): 0; Gene Names: mmoX; EC: 1.14.13.25

## Entity ID: 3
* Molecule: Methane monooxygenase component A gamma chain
* Chains: C, H, M, R
* Sequence Length: 170
* Organism: Methylococcus capsulatus (strain ATCC 33009 / NCIMB 11132 / Bath)
* Details: Mutation(s): 0; Gene Names: mmoZ; EC: 1.14.13.25

## Entity ID: 4
* Molecule: Methane monooxygenase regulatory protein B
* Chains: D, I, N, S
* Sequence Length: 141
* Organism: Methylococcus capsulatus (strain ATCC 33009 / NCIMB 11132 / Bath)
* Details: Mutation(s): 0; Gene Names: mmoB

Only entity id:4 of the lengths found on pdb website for complex `4GAM` seems to have a ligand or receptor length that matches what is found in the numpy array provided.

In [35]:
# determine if there is a matching between the expected dims for the label and the actual value
all_test_r_vertex_lens = [i["r_vertex"].shape[0] for i in test[1][:]]
all_test_l_vertex_lens = [i["l_vertex"].shape[0] for i in test[1][:]]
all_test_label_lens = [i["label"].shape[0] for i in test[1][:]]

for i in zip(all_test_r_vertex_lens, all_test_l_vertex_lens, all_test_label_lens):
  print("match: {} actual: {} expected: {}".format(i[0]*i[1]==i[2], i[2], i[0]*i[1]))

match: False actual: 221095 expected: 229284
match: False actual: 279472 expected: 300048
match: False actual: 59073 expected: 60416
match: False actual: 146648 expected: 149193
match: False actual: 447036 expected: 514800
match: False actual: 122940 expected: 132987
match: False actual: 40177 expected: 41942
match: False actual: 93881 expected: 96135
match: False actual: 37018 expected: 41328
match: False actual: 30464 expected: 32130
match: False actual: 636126 expected: 638880
match: False actual: 19816 expected: 20207
match: False actual: 42443 expected: 43308
match: False actual: 59766 expected: 62054
match: False actual: 6526 expected: 6555
match: False actual: 633604 expected: 647094
match: False actual: 63975 expected: 64930
match: False actual: 93394 expected: 101010
match: False actual: 7100 expected: 7920
match: False actual: 75558 expected: 76970
match: False actual: 5339 expected: 6324
match: False actual: 177517 expected: 179280
match: False actual: 41401 expected: 44200


# Exploration of Train file

In [20]:
from collections import Counter
train_path_to_cpkl_file = "/home/mg/workspace_mg/Interaction_Predictions/fout_protein_interface_prediction_using_graph_convolutional_networks/pl_data/train.cpkl"
train = load_cpkl(path_to_cpkl_file=train_path_to_cpkl_file)

# Train example 1 on Complex: `1BJ1`

In [22]:
complex_code = train[1][10]["complex_code"]
print("r_vertex shape: {} for complex code: {}".format(train[1][10]["r_vertex"].shape, complex_code))
print("l_vertex shape: {} for complex code: {}".format(train[1][10]["l_vertex"].shape, complex_code))
print("r_edge shape: {} for complex code: {}".format(train[1][10]["r_edge"].shape, complex_code))
print("l_edge shape: {} for complex code: {}".format(train[1][10]["l_edge"].shape, complex_code))
print("r_hood_indices shape: {} for complex code: {}".format(train[1][10]["r_hood_indices"].shape, complex_code))
print("l_hood_indices shape: {} for complex code: {}".format(train[1][10]["l_hood_indices"].shape, complex_code))
print("label shape: {} for complex code: {}".format(train[1][10]["label"].shape, complex_code))
print("expected dimensions for axis 1 for 'label' numpy array: {}".format(train[1][10]["r_vertex"].shape[0]*train[1][10]["l_vertex"].shape[0]))
print("actual dimensions for axis 1 for 'label' numpy array: {}".format(train[1][10]["label"].shape[0]))
print("label distribution: {}".format(Counter(train[1][10]["label"][:,2])))

r_vertex shape: (431, 70) for complex code: 1BJ1
l_vertex shape: (189, 70) for complex code: 1BJ1
r_edge shape: (431, 20, 2) for complex code: 1BJ1
l_edge shape: (189, 20, 2) for complex code: 1BJ1
r_hood_indices shape: (431, 20, 1) for complex code: 1BJ1
l_hood_indices shape: (189, 20, 1) for complex code: 1BJ1
label shape: (1111, 3) for complex code: 1BJ1
expected dimensions for axis 1 for 'label' numpy array: 81459
actual dimensions for axis 1 for 'label' numpy array: 1111
label distribution: Counter({-1: 1010, 1: 101})


# complex_code `1BJ1`
Searched for more information on pdb website https://www.rcsb.org/structure/1BJ1 to confirm length for receptor and ligand structures.

## Entity ID: 1
* Molecule: Fab fragment, light chain
* Chains: L, J
* Sequence Length: 214
* Organism: NA
* Details: Mutation(s): 0

## Entity ID: 2
* Molecule: Fab fragment, heavy chain	
* Chains: H, K
* Sequence Length: 231
* Organism: NA
* Details: Mutation(s): 0

## Entity ID: 3
* Molecule: Vascular endothelial growth factor A
* Chains: V, W
* Sequence Length: 102
* Organism: Homo Sapiens
* Details: Mutation(s): 0; Gene Names: VEGFA (VEGF)

None of the lengths found on pdb website for complex `1BJ1` seem to have a ligand or receptor length that matches what is found in the numpy array provided.

In [36]:
# determine if there is a matching between the expected dims for the label and the actual value
all_train_r_vertex_lens = [i["r_vertex"].shape[0] for i in train[1][:]]
all_train_l_vertex_lens = [i["l_vertex"].shape[0] for i in train[1][:]]
all_train_label_lens = [i["label"].shape[0] for i in train[1][:]]

for i in zip(all_train_r_vertex_lens, all_train_l_vertex_lens, all_train_label_lens):
  print("match: {} actual: {} expected: {}".format(i[0]*i[1]==i[2], i[2], i[0]*i[1]))

match: False actual: 1683 expected: 66970
match: False actual: 704 expected: 11973
match: False actual: 836 expected: 71668
match: False actual: 1584 expected: 63332
match: False actual: 1518 expected: 35100
match: False actual: 1012 expected: 474138
match: False actual: 847 expected: 15756
match: False actual: 1496 expected: 63336
match: False actual: 1056 expected: 85500
match: False actual: 1694 expected: 80960
match: False actual: 1111 expected: 81459
match: False actual: 693 expected: 424886
match: False actual: 737 expected: 45032
match: False actual: 605 expected: 14420
match: False actual: 1155 expected: 41503
match: False actual: 704 expected: 30888
match: False actual: 836 expected: 11397
match: False actual: 781 expected: 70980
match: False actual: 913 expected: 55728
match: False actual: 1100 expected: 17810
match: False actual: 385 expected: 38220
match: False actual: 935 expected: 23973
match: False actual: 825 expected: 26028
match: False actual: 990 expected: 79786
matc